src/js/components/tasks/submit.tsx

```diff
@@ -15,17 +15,19 @@ import {
 import { OBJECT_TYPES } from '@/utils/constants';
 
 interface Props {
-  taskId: string;
-  taskName: string;
-  taskDiffUrl: string | null;
+  instanceId: string;
+  instanceName: string;
+  instanceDiffUrl: string | null;
+  instanceType: string;
```

Comment:
Let's make this more strict.

```suggestion
  instanceType: 'task' | 'project';
```